applications/registration-api/src/main/java/no/dcat/service/ApiCatalogHarvesterService.java

```diff
@@ -0,0 +1,154 @@
+        this.apiCatalogRepository = apiCatalogRepository;
+        this.harvestQueue = harvestQueue;
+        this.apiCatClient = apiCatService.getClient();
+    }
+
+    //TODO: Also trigger this every 24h ? (there exists a spring thing to do that)
+    @PostConstruct
+    public void initQueue() {
+        logger.info("Survived to PostConstruct");
+        //Get all the catalogs
+        List<ApiCatalog> allApiCatalogs = apiCatalogRepository.findAll().getContent();
+
+
+        if (allApiCatalogs == null || allApiCatalogs.size() == 0) {
+            //Add the single hardcoded API Catalog we know now.
+            allApiCatalogs = new ArrayList<>();
+            ApiCatalog hardcodedCatalog = new ApiCatalog();
+            hardcodedCatalog.setHarvestSourceUri("https://raw.githubusercontent.com/brreg/openAPI/master/specs/API-catalog.ttl");
+            allApiCatalogs.add(hardcodedCatalog);
+        }
+
+        //Add them to the queue
+        for (ApiCatalog catalog : allApiCatalogs) {
+            QueuedTask task = new QueuedTaskImpl();
+            ((QueuedTaskImpl) task).description = "A catalog";
+            try {
+                URL catalogURL = new URL(catalog.getHarvestSourceUri());
+                ((QueuedTaskImpl) task).urlLocation = catalogURL;
+                ((QueuedTaskImpl) task).catalog = catalog;
+                harvestQueue.addTask(task);
+            } catch (MalformedURLException malf) {
+                logger.error("Failed to parse {} as URL. Continuing to next catalog.", catalog.getHarvestSourceUri());
+            }
+        }
+    }
+
+    protected void doHarvestSingleCatalog(URL urlToCatalog, ApiCatalog originCatalog) {
+        logger.info("HarvestSingleCatalog called! url " + urlToCatalog + " origin catalog: " + originCatalog);
+        Status status = new Status();
+        status.setSuccess(true);
+
+        try {
+            BufferedReader reader = new BufferedReader(
+                new InputStreamReader(urlToCatalog.openStream()));
+            final Model model = ModelFactory.createDefaultModel();
+            model.read(reader, null, "TURTLE");//Base and lang is just untested dummy values
+
+            List<String> allApiUrlsForThisCatalog = new ArrayList<String>();
+            StmtIterator iterator = model.listStatements();
+            while (iterator.hasNext()) {
+                Statement st = iterator.next();
+                if (st.getPredicate().toString().contains("dcat#dataset")) {
+                    allApiUrlsForThisCatalog.add(st.getObject().toString());
+                }
+            }
+
+            for (String apiUrl : allApiUrlsForThisCatalog) {
+                ApiRegistration apiRegistrationData = new ApiRegistration();
+                apiRegistrationData.setApiSpecUrl(apiUrl);
+
+                //ConvertTo Reference API
+                OpenAPI openAPI = apiCatClient.convert(apiUrl, "");//empty string is api-spec, unknown what this is.
```

Comment:
ok, da trenger vi bedre dokumentasjon :D eller klient lib skulle ha to metoder (eller tre)

Reply:
Er enig med det :)